src/libpriv/rpmostree-importer.cxx

```diff
@@ -515,6 +515,7 @@ compose_filter_cb (OstreeRepo         *repo,
         case G_FILE_TYPE_SYMBOLIC_LINK:
           break;
         default:
+          g_warning ("Not importing spurious content at %s", path);
```

Comment:
In pathological cases, printing one line per file could flood the logs.  In the ostree code [around here](https://github.com/ostreedev/ostree-rs-ext/blob/a6b23c2249652dce5573283161d15c5368fd7860/lib/src/tar/write.rs#L135) we gather a mapping of (toplevel path) :arrow_right: count, e.g. `/var` :arrow_right: 20 and `/opt` :arrow_right: 10.

Another approach might be to print at most e.g. 5 lines per package?

One interesting thing about the current ostree code is that it stores the above metadata as part of the commit, which means it's not ephemeral.  This is a bit like what `cargo build` does with compiler warnings.  Every time you invoke `cargo build`, it re-prints the compiler warnings even if the code didn't change.  To me, this is a *huge* ergonomic improvement over `gcc` for example.

Reply:
I mostly needed this for a sanity check on `vagrant` (where I misjudged a ghost entry).

You are right, this can be improved for better UX.
As a reality check this triggers exactly twice on the full FCOS package set (and one case is already fixed upstream).

For now, I'm fine demoting this to `g_debug` so that it can't spam unless explicitly asked for.

Reply:
Relatedly we could try using [glib structured logging](https://developer-old.gnome.org/glib/stable/glib-Message-Logging.html#g-log-structured).